torch/_prims/utils.py

```diff
@@ -3,19 +3,37 @@
 from numbers import Number
 from typing import Any, Union, Sequence, Optional, Callable, Dict, Tuple, List
 from functools import reduce
+import threading
 
 import torch
+from torch.fx import Node
 
 ShapeType = Union[torch.Size, List[int], Tuple[int, ...]]
 StrideType = Union[List[int], Tuple[int, ...]]
 DimsType = Union[int, List[int], Tuple[int, ...]]
 
 
-class TensorMeta(object):
+class TensorMeta_Meta(type):
```

Comment:
Is there any good explanation why we need a metaclass for TensorMeta?

Reply:
Idk about "good" -- there's probably a better way to achieve these, but it lets us add a thread local class property